_posts/2016-08-05-lambda.markdown

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: Scaling with AWS Lambda
+date: 2016-08-05 20:35:04
+description: Scale your applications with AWS Lambda without having to think about it. This is a case study from a migration done at Lost My Name.
+---
+At [Lost My Name](https://www.lostmy.name/) I'm responsible for the reliability and performance of the platform. This platform is the core of the business as it enables the creatives to publish their work. This work is what makes Lost My Name's products so magical and popular.
+
+As of writing, the team is preparing for another Q4. This will be my second time going through this period at Lost My Name after been through it twice at [Boomf](https://boomf.com/). The experience is unique as you see the sales charts go up and up.
+
+Increased sales means increased prints we need to deliver to our printhouses. Every single one of hese prints goes though the platform to generate the personalised content at high resolution. Increased sales also means increased previews which also goes through the platform.
+
+In this blog post, I'll talk about a specific piece of one of our books, [The Incredible Intergalactic Journey Home](https://www.lostmy.name/personalized-products/the-journey-home-book).
+
+### The current situation
+
+For one of the spreads in the book - which exists out of 2 pages - we request your address. We then generate a page which consists of satellite imagery data to show kids their house and the area around their house. To do this, we use the [Bing Imagery API](https://msdn.microsoft.com/en-us/library/ff701721.aspx).
+
+To generate one page at print quality, we need to compose 4 Bing images at maximum resolution (2000x1500px). To have consistency - and some other non relevant reasons - between preview and print, we generate the same print quality image for the preview which we then resize before serving it to the user.
+
+This means that for every user that requests a book preview, we need to download 8 (1 spread, 2 pages) images, stitch them together and resize the image afterwards. All this work is quite CPU intensive and can't be cached (every request is unique due to it's specific coordinates).
```

Comment:
"(every request is unique due to it's specific coordinates)" => "(every request is unique due to the specific coordinates)"